src/pages/verify-email/[token].tsx

```diff
@@ -0,0 +1,23 @@
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+import { api } from "~/utils/api";
+import VerifyEmailError from "~/components/verify-email/VerifyEmailError";
+import VerifyEmailSuccess from "~/components/verify-email/VerifyEmailSuccess";
+
+const VerifyEmail = () => {
+  const router = useRouter();
+  const { token } = router.query;
+  const verifyEmailMutation = api.auth.verifyEmail.useMutation();
+
+  useEffect(() => {
+    if(token) {
+      verifyEmailMutation.mutate({ token: String(token) });
+    }
+  }, [token, router]);
+
+  if(verifyEmailMutation.isError) return <VerifyEmailError />
+  if(verifyEmailMutation.isSuccess) return <VerifyEmailSuccess />
```

Comment:
I think it's better to have `VerifyEmailSuccess` as the default returned component. Basically swap this with the loading fragment and change the condition to `verifyEmailMutation.isLoading`

Reply:
I tried your method but found a minor issue. I see success component initially flashing when visiting the page and then goes to error (This is assuming the verify URL was bad or token was invalid).

I think initially while the component is mounting, the state of verifyEmailMutation is neither isError, isSuccess, or isLoading, so the default return of success component is shown briefly until the state changes to isLoading.

Reply:
Ah, I see. If it works as it is without use an `useEffect()` hook, it's probably for the best.